V1.2 delete important dates: Logic & Test

ZhiWei1010 opened the pull request:
This commit allows for important dates to be deleted for patients.

- Add new DeleteImportantDateCommand
- Add new DeleteImportantDateCommandParser
- Add new DeleteImportantDateCommandTest
- Add new DeleteImportantDateCommandParserTest
- Update User Guide to include guide on DeleteImportantDateCommand
- Update Developer guide to include guide on DeleteImportantDateCommand

To be done:  
- [X] format of command
- [X] testing
- [X] User Guide
- [X] Developer Guide

Fixes #68 
Fixes #69 
Fixes #7

ZhiWei1010 committed:
Logic: Add new DeleteImportantDateCommand

Add a new DeleteImportantDateCommand class to handle the deletion of
important dates by patient id and important date id.

Teach app to accept deleteID as a new command word and return a
new DeleteImportantDateCommand if format is correct, else throw an
exception.

ZhiWei1010 committed:
Logic: Teach model to delete all ImportantDates of Patient

Update model to delete all important dates of patient instead of by
event index.

ZhiWei1010 committed:
Update check style to match guidelines

NeoHW commented on src/main/java/seedu/address/logic/commands/DeleteImportantDateCommand.java:
```diff
@@ -0,0 +1,90 @@
+ */
+public class DeleteImportantDateCommand extends Command {
+
+    public static final String COMMAND_WORD = "deleteID";
```
could consider following naming convention 
e.g. adding important dates : `adde`
can consider something similar to `deletee`?

ZhiWei1010 replied:
Changed command to `deletee` for consistency

NeoHW commented on src/main/java/seedu/address/logic/commands/DeleteImportantDateCommand.java:
```diff
@@ -0,0 +1,90 @@
+            + COMMAND_WORD + " 1 "
+            + "e/ 1 ";
+
+    public static final String MESSAGE_DELETE_IMPORTANT_DATE_SUCCESS = "Deleted Important Dates for Patient: %1$s";
```
Deleting one date: Delete Important Date~s~

ZhiWei1010 replied:
Good catch! Fixed.

NeoHW commented on src/main/java/seedu/address/logic/parser/AddressBookParser.java:
```diff
@@ -85,6 +86,10 @@ public Command parseCommand(String userInput) throws ParseException {
 
+        case DeleteImportantDateCommand.COMMAND_WORD:
+            return new DeleteImportantDateCommandParser().parse(arguments);
+
```
Extra line?

ZhiWei1010 replied:
Good Catch. Fixed

NeoHW commented on src/main/java/seedu/address/logic/parser/CliSyntax.java:
```diff
@@ -12,5 +12,6 @@ public class CliSyntax {
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
     public static final Prefix PREFIX_DATETIME = new Prefix("d/");
+    public static final Prefix PREFIX_IMPORTANT_DATE = new Prefix("e/");
```
it might be confusing for user to have adding of ID be n/ d/ 
then introduce deleting to use e/

ZhiWei1010 replied:
The idea is for user to quickly delete the event via event ID instead of typing the whole date to delete, so I think this is fine. It is also consistent with the convention of adding important date 'adde'. Should be ok for user to learn this. Could be discussed further with input from others!

NeoHW reviewed (changes requested):
consider design and additional changes

ZhiWei1010 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2103-F09-2/tp into v1.2-delete-important-dates

ZhiWei1010 committed:
Logic: Update DeleteImportantDateCommand to delete by event ID

Update COMMAND_WORD for DeleteImportantDateCommand to be 'deletee' for
consistency with 'adde'.

Update MESSAGE_USUAGE to stay consistent with new format.

Remove extra empty line from
AddressBookParser#parseCommand(String userInput).

ZhiWei1010 committed:
Merge branch 'master' into v1.2-delete-important-dates

ZhiWei1010 committed:
Logic: Update equals() method and toString() method to work correctly

Add a check for targetEventIndex in DeleteImportantDateCommand#Equals().

Add a new field of targetEventIndex for
DeleteImportantDateCommand#toString().

ZhiWei1010 committed:
Logic: Update constructor in DeleteImportantDateCommand

Add requiredAllNonNull check for both parameters of constructor.

ZhiWei1010 committed:
Test: Add tests for DeleteImportantCommand

Create new DeleteImportantCommandTest class to test
DeleteImportantCommand.

Add relevant tests to ensure that DeleteImportantCommand works as
expected.

ZhiWei1010 committed:
Test: Add more tests for DeleteImportantDateCommand

Add Test for Invalid Event Id but valid Patient Id.

Add new test case in equals() method

ZhiWei1010 committed:
Test: Add new DeleteImportantDateCommandParserTest class

Test to ensure DeleteImportantDateCommandParser works as expected.

ZhiWei1010 committed:
Documentation: Update User Guide and Developer Guide

Add new section for Delete Important Date command in User Guide,
teaching users how to use the command, includes some examples as well.

Add new section for Deleting Important Date command in Developer Guide.

choowengyan reviewed (approved):
LGTM!

NeoHW reviewed (approved):
code lgtm, ui tested and working
